torch/_meta_registrations.py

```diff
@@ -323,9 +323,8 @@ def meta_index_Tensor(self, indices):
     return self.new_empty(before_shape + replacement_shape + after_shape)
 
 
-@torch.library.impl(meta_lib, "addbmm")
```

Comment:
Doesn't this explain why the test is failing?

Reply:
That was indeed wrong (corrected) but was not the cause of the error. The thing is that somewhere the `out` parameter (which is a tuple) is being transformed into several kwargs, each with the name of the corresponding output. This is wrong, and I have not been able to track down where it happens, but it seems related to the contextmanager `enable_torch_function_mode(inner)`...

To see the backtrace of the failures see https://github.com/pytorch/pytorch/runs/7062769044?check_suite_focus=true

Reply:
The function in question here is being registered as an ATen op directly in the dispatcher. Theseget invoked by the function: parseIValuesToPyArgsKwargs

```
  // Populate positional arguments
  for (const auto idx : c10::irange(positional_default_start)) {
    PyTuple_SET_ITEM(
        args.ptr(), idx, schemaAwareToPyObject(idx).release().ptr());
  }
      
  // Populate keyword arguments
  for (const auto idx : c10::irange(kwarg_only_start, arguments.size())) {
    // But don't populate default keyword arguments
    if (is_default(idx))
      continue;
    const auto& arg = schema.arguments()[idx];
    kwargs[py::cast(arg.name())] = schemaAwareToPyObject(idx);
  }   
  return std::make_pair(std::move(args), std::move(kwargs));
```

There is no special handling for out arguments. The dispatcher just slaps them exactly as they look in native_functions.yaml

I'm not convinced I want to change this behavior. But that means we will need two different implementations of out handling, one for refs (which are Python visible) and another for the ATen registrations (which will see them unpacked.)

Reply:
Couldn't this be solved just by packing the `out` parameters back into a tuple in that context manager before feeding them back to the ref?

Reply:
Yes, but the main point is you need two versions of the function, one to be a ref and one to actually register to the dispatcher